modules/reindex/src/main/java/org/elasticsearch/index/reindex/BulkByScrollTask.java

```diff
@@ -50,15 +58,39 @@
     private final AtomicLong versionConflicts = new AtomicLong(0);
     private final AtomicLong retries = new AtomicLong(0);
     private final AtomicLong throttledNanos = new AtomicLong();
+    /**
+     * The number of requests per second to which to throttle the request that this task represents. The other variables are all AtomicXXX
+     * style variables but there isn't an AtomicFloat so we just use a volatile.
+     */
+    private volatile float requestsPerSecond;
+    private final AtomicReference<DelayedPrepareBulkRequest> delayedPrepareBulkRequest = new AtomicReference<>();
```

Comment:
Same here with naming, we have `delayedPrepareBulkRequest` and `delayPrepareBulkRequest` which are confusing when vars and functions are named so similarly